hw10_motion_sensor/main.go

```diff
@@ -1,5 +1,60 @@
 package main
 
+import (
+	"crypto/rand"
+	"fmt"
+	"math/big"
+	"time"
+)
+
 func main() {
-	// Place your code here.
+	rawDataChan := make(chan int)
+	processedDataChan := make(chan float64)
+	go readSensorData(rawDataChan)
+	go processDataUsingAverage(rawDataChan, processedDataChan)
+
+	for data := range processedDataChan {
+		fmt.Printf("Processed data: %.2f\n", data)
+	}
+}
+
+// Функция имитации считывания данных с сенсора и передачи их в канал.
+func readSensorData(dataChan chan<- int) {
```

Comment:
В go хорошей практикой считается закрывать канал там, где он создается. Поэтому в данном случае было бы правильнее если бы эта функция выгладила вот так
```
func readSensorData() <-chan int {
	dataChan := make(chan int)
	go func() {
		defer close(dataChan)
		for start := time.Now(); time.Since(start) < time.Minute; {
			n, err := rand.Int(rand.Reader, big.NewInt(100))
			if err != nil {
				fmt.Println("Random number generation error:", err)
				return
			}
			dataChan <- int(n.Int64())
			time.Sleep(time.Second)
		}
	}()
	return dataChan
}
```